webapp/components/survey/CategorySelector/CategorySelector.js

```diff
@@ -71,24 +61,33 @@ const CategorySelector = (props) => {
           {i18n.t('common.manage')}
         </button>
       )}
-      {showCategoriesPanel && !editedCategory && (
+      {showCategoriesPanel && !categoryToEdit && (
         <PanelRight
           width="100vw"
           onClose={() => setShowCategoriesPanel(false)}
           header={i18n.t('appModules.categories')}
         >
-          <CategoryList canSelect selectedItemUuid={categoryUuid} onSelect={onChange} />
+          <CategoryList
+            canSelect
+            selectedItemUuid={categoryUuid}
+            onSelect={onChange}
+            onEdit={(categoryToEditSelected) => {
+              setCategoryToEdit(categoryToEditSelected)
```

Comment:
why not just the uuid of the category? anyway when you edit it, you fetch it again, or not?

Reply:
no, if you pass the entire category to CategoryDetails, it won't be fetched again

Reply:
but dont you need to fetch at least the first level?

Reply:
yes, it will be done in the next pull request, together with the handling of the levels